src/store/modules/courses.js

```diff
@@ -1,74 +1,81 @@
+import axios from 'axios';
+
 const courses = {
   namespaced: true,
   state() {
     return {
       courses: [],
-      courseById: null
+      courseId: null
     };
   },
   getters: {
-    error(state) {
-      return state.error;
-    },
-    isLoading(state) {
-      return state.isLoading;
-    },
     courses(state) {
       return state.courses;
     },
-    courseById(state) {
-      return state.courseById;
+    courseId(state) {
+      return state.courseId;
     }
   },
   mutations: {
-    updateCourses(state, payload) {
-      state.courses = payload;
+    updateCourses(state, allCourses) {
+      state.courses = allCourses;
     },
-    getCourseById(state, payload) {
-      state.courseById = payload;
+    getCourse(state, course) {
```

Comment:
change mutation name to updateCourseById

Reply:
done